lib/Sema/CSDiagnostics.cpp

```diff
@@ -5515,6 +5533,55 @@ bool MissingArgumentsFailure::diagnoseClosure(const ClosureExpr *closure) {
   return true;
 }
 
+bool MissingArgumentsFailure::diagnoseMissingResultBuilderElement() const {
+  auto &ctx = getASTContext();
+
+  // Only handle a single missing argument in an empty builder for now. This
+  // should be the most common case though since most builders support N >= 1
+  // elements.
+  if (SynthesizedArgs.size() != 1)
+    return false;
+
+  auto *call = getAsExpr<CallExpr>(getRawAnchor());
+  if (!call || !call->isImplicit() || !call->getArgs()->empty())
+    return false;
+
+  auto *UDE = dyn_cast<UnresolvedDotExpr>(call->getFn());
+  if (!UDE || !UDE->isImplicit())
+    return false;
+
+  auto overload = getCalleeOverloadChoiceIfAvailable(getLocator());
+  if (!overload)
+    return false;
+
+  auto *decl = overload->choice.getDeclOrNull();
+  if (!decl || decl->getBaseName() != ctx.Id_buildBlock)
```

Comment:
There is a method we use already to determine that based on expression itself - `isResultBuilderMethodReference` all you'd need to do is check whether it's also `Id_buildBlock` specifically as well. 

For completeness you might also want to cover `Id_buildPartialBlock` I think.

Reply:
Ah nice, updated to use `isResultBuilderMethodReference`. `buildPartialBlock` is only used for non-empty cases I believe, so isn't applicable at least to the current case being handled here.